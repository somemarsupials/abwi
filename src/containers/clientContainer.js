import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchClient } from '../actions';
import { Page } from '../components/clients';


export class ClientContainer extends Component {
  async componentDidMount() {
    this.props.fetchClient(this.props.match.params.id);
  };

  render() {
    return (
      <Page
        client={this.props.client} 
        error={this.props.error}
        fetching={this.props.fetching}
      />
    );
  };
};

function mapStateToProps(state) {
  return { 
    client: state.client.data,
    error: state.client.error,
    fetching: state.client.fetching,
  };
};

function mapDispatchToState(dispatch) {
  return {
    fetchClient: (id) => dispatch(fetchClient(id)),
  };
};

export default 
  connect(mapStateToProps, mapDispatchToState)(ClientContainer);
