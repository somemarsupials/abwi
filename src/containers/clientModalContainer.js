import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ClientModal } from '../components/modals';
import { toggleClientModal, nameChange } from '../actions/clientModal';
import { createClient } from '../actions/clientModal';

class ClientModalContainer extends Component {
  render() {
    return (
      <ClientModal {...this.props} />
    );
  };
};

function mapStateToProps(state) {
  return state.clientModal;
};

function mapDispatchToState(dispatch) {
  return {
    toggle: () => dispatch(toggleClientModal()),
    nameChange: string => dispatch(nameChange(string)),
    onSubmit: (params) => dispatch(createClient(params)),
  };
};

export default 
  connect(mapStateToProps, mapDispatchToState)(ClientModalContainer)
