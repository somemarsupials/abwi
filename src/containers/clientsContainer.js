import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchClients } from '../actions';
import { Page } from '../components/allClients';


export class ClientsContainer extends Component {
  async componentDidMount() {
    this.props.fetchClients();
  };

  render() {
    return (
      <Page
        clients={this.props.clients} 
        error={this.props.error}
        fetching={this.props.fetching}
      />
    );
  };
};

function mapStateToProps(state) {
  return { 
    clients: state.clients.data,
    error: state.clients.error,
    fetching: state.clients.fetching,
  };
};

function mapDispatchToState(dispatch) {
  return {
    fetchClients: () => dispatch(fetchClients()),
  };
};

export default 
  connect(mapStateToProps, mapDispatchToState)(ClientsContainer);

