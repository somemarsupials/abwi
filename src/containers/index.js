export { default as ProjectsContainer } from './projectsContainer';
export { default as ProjectContainer } from './projectContainer';
