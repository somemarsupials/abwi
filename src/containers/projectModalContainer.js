import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ProjectModal } from '../components/modals';

import { toggleProjectModal, titleChange, descriptionChange, clientChange } 
  from '../actions/projectModal';

import { createProject } from '../actions/projectModal';

class ProjectFormContainer extends Component {
  componentWillMount() {
    this.props.descriptionChange(null)
    this.props.clientChange(null)
  };

  render() {
    return (
      <ProjectModal {...this.props} />
    );
  };
};

function mapStateToProps(state) {
  return state.projectModal;
};

function mapDispatchToState(dispatch) {
  return {
    toggle: () => dispatch(toggleProjectModal()),
    titleChange: string => dispatch(titleChange(string)),
    descriptionChange: string => dispatch(descriptionChange(string)),
    clientChange: string => dispatch(clientChange(string)),
    onSubmit: (name, params) => dispatch(createProject(name, params)),
  };
};

export default 
  connect(mapStateToProps, mapDispatchToState)(ProjectFormContainer)
