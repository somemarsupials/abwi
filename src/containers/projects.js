import React, { Component } from 'react';
import { connect } from 'react-redux';

import { projectActions } from '../actions';
import { ProjectsOverview } from '../components/projects';


export class Projects extends Component {
  async componentDidMount() {
    let { api, success, fail } = this.props;
    this.props.fetch(api, success, fail);
  };

  render() {
    return (
      <ProjectsOverview
        projects={this.props.projects} 
        error={this.props.error}
      />
    );
  };
};


function mapStateToProps(state) {
  return { 
    projects: state.projects.data,
    error: state.projects.error,
  };
};


function mapDispatchToProps(dispatch) {
  let { fetchedProjects, projectFetchFailed } = projectActions;
  return {
    success: (projects) => {
      dispatch(fetchedProjects(projects))
    },
    fail: (error) => {
      dispatch(projectFetchFailed(error))
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Projects);
