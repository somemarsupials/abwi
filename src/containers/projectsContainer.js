import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchProjects, toggleCreateProjectModal } 
  from '../actions/allProjects';

import { ProjectsPage } from '../components/allProjects';


export class ProjectsContainer extends Component {
  async componentDidMount() {
    this.props.fetchProjects();
  };

  render() {
    return (
      <ProjectsPage {...this.props} />
    );
  };
};

function mapStateToProps(state) {
  return { 
    projects: state.projects.data,
    error: state.projects.error,
    fetching: state.projects.fetching,
    createModal: state.projects.createModal,
  };
};

function mapDispatchToState(dispatch) {
  return {
    fetchProjects: () => dispatch(fetchProjects()),
    toggleCreateModal: () => dispatch(toggleCreateProjectModal(true))
  };
};

export default 
  connect(mapStateToProps, mapDispatchToState)(ProjectsContainer);
