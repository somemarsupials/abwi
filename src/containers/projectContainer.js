import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchProject } from '../actions/projects';
import { ProjectPage } from '../components/projects';


export class ProjectContainer extends Component {
  async componentDidMount() {
    this.props.fetchProject(this.props.match.params.id);
  };

  render() {
    return (
      <ProjectPage
        project={this.props.project} 
        error={this.props.error}
        fetching={this.props.fetching}
      />
    );
  };
};

function mapStateToProps(state) {
  return { 
    project: state.project.data,
    error: state.project.error,
    fetching: state.project.fetching,
  };
};

function mapDispatchToState(dispatch) {
  return {
    fetchProject: (id) => dispatch(fetchProject(id)),
  };
};

export default 
  connect(mapStateToProps, mapDispatchToState)(ProjectContainer);
