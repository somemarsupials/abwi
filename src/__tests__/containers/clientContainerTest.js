import React from 'react';
import { shallow } from 'enzyme';

import { ClientContainer } from '../../containers/clientContainer';

describe('Client', () => {
  let props, client, mockFetch;

  beforeEach(() => {
    props = { fetchClient: jest.fn(), match: { params: { id: 5 } } };
  });

  describe('when mounted', () => {
    beforeEach(() => {
      client = shallow(<ClientContainer {...props} />);
    });

    it('calls fetch function', () => {
      expect(props.fetchClient).toHaveBeenCalledWith(5);
    });
  });


  describe('when rendering', () => {
    let mockFetch, clientPage;

    beforeEach(() => {
      Object.assign(props, { 
        client: { obj: 'props' }, 
        error: 'error', 
        fetching: false,
      });
      client = shallow(<ClientContainer {...props} />);
    });

    it('creates ClientPage', () => {
      expect(client).toMatchSnapshot();
    });
  });
});
