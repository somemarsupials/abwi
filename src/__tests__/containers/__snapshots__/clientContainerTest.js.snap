// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client when rendering creates ClientPage 1`] = `
<Component
  client={
    Object {
      "obj": "props",
    }
  }
  error="error"
  fetching={false}
/>
`;
